src/jit/importer.cpp

```diff
@@ -3208,7 +3201,7 @@ GenTreePtr      Compiler::impIntrinsic(CORINFO_CLASS_HANDLE     clsHnd,
         {
             NO_WAY("JIT must expand the intrinsic!");
         }
-        else if (IsIntrinsicImplementedByUserCall(intrinsicID))
+        else if ((retNode->gtFlags & GTF_CALL) != 0)
```

Comment:
@CarolEidt  I will add a comment.
@sivarv As commented in the PR, IsIntrinsicImplementedByUserCall check should be for math intrinsics only. For instance, any other non-math intrinsic like CORINFO_INTRINSIC_InitializeArray should not be even under a consideration. Here IsIntrinsicImplementedByUserCall returns true since it is just not IsTargetIntrinsic. This mis-led us to interpret CORINFO_INTRINSIC_InitializeArray  is to be implemented by a user call.  Ideally we should refactor this API for more reasonable way, which I haven't incorporated in this change on purpose.